pkg/controllers/configobservation/clusterversion/observe_clustercapabilities.go

```diff
@@ -0,0 +1,43 @@
+package clusterversion
+
+import (
+	"os"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/controllers/configobservation"
+)
+
+func ObserveClusterCapabilities(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, _ []error) {
+	listers := genericlisters.(configobservation.Listers)
+	errs := []error{}
+
+	clusterVersionConfig, err := listers.ClusterVersionLister.Get("version")
+	if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	for _, capability := range clusterVersionConfig.Status.Capabilities.EnabledCapabilities {
+		if capability == configv1.ClusterVersionCapabilityConsole {
+			recorder.Eventf("ObserveClusterCapabilities", "Console capability enabled, restarting cluster authentication operator")
+			klog.Infof("Console capability enabled, restarting cluster authentication operator")
+			f, err := os.OpenFile("/tmp/terminate", os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
```

Comment:
That's an interesting pattern. Basically we're working around being able to `cancel()` the global context from the inside of the operator :thinking:

Reply:
yeah, the global context/cancel function created in the library isn't exposed to the operator itself so i couldn't just cancel it, at least i didn't see a way...i'd certainly have preferred to have just done that.  But i also didn't want to go refactoring openshift-library to make it possible.